app/src/main/java/com/infomaniak/drive/utils/Utils.kt

```diff
@@ -366,30 +366,34 @@ object Utils {
 
     fun downloadAsOfflineFiles(context: Context, folderId: Int, userDrive: UserDrive = UserDrive(), onSuccess: () -> Unit) =
         liveData {
-            val workManager = WorkManager.getInstance(context)
-            val inputData = workDataOf(
-                BulkDownloadWorker.FOLDER_ID to folderId,
-                BulkDownloadWorker.USER_ID to userDrive.userId,
-                BulkDownloadWorker.DRIVE_ID to userDrive.driveId,
-            )
-            val networkType = if (AppSettings.onlyWifiSync) NetworkType.UNMETERED else NetworkType.CONNECTED
-            val constraints = Constraints.Builder()
-                .setRequiredNetworkType(networkType)
-                .setRequiresStorageNotLow(true)
-                .build()
-            val downloadRequest = OneTimeWorkRequestBuilder<BulkDownloadWorker>()
-                .addTag(BulkDownloadWorker::class.java.toString())
-                .setInputData(inputData)
-                .setConstraints(constraints)
-                .setExpeditedIfAvailable()
-                .build()
-
-            workManager.enqueueUniqueWork(BulkDownloadWorker.TAG, ExistingWorkPolicy.APPEND_OR_REPLACE, downloadRequest)
+            enqueueBulkDownloadWorker(context, folderId, userDrive)
 
             onSuccess.invoke()
             emit(FileResult(isSuccess = true))
         }
 
+    fun enqueueBulkDownloadWorker(context: Context, folderId: Int, userDrive: UserDrive = UserDrive()) {
```

Comment:
Check if folder has `marked offline files` before to call this method

Reply:
Done